### PR TITLE
fix: Update library for IJ 2023.2

### PR DESCRIPTION
**Pull request by martinszuc**
This is PR aims to add IntelliJ IDEA 2023.2 to the library and fix issues found in epic: https://github.com/redhat-developer/intellij-common-ui-test-library/issues/187

### CODE REVIEW

File: src/test-project/src/test/java/com/redhat/devtools/intellij/commonuitest/fixtures/test/dialogs/project_manipulation/NewProjectDialogTest.java [hunk 1]
```diff
@@ -333,25 +333,29 @@ public void cancelButtonTest() {
 
     @Test
     public void setProjectSdkIfAvailableTest() {
-        newProjectFirstPage.setProjectSdkIfAvailable("8");
+        newProjectFirstPage.setProjectSdkIfAvailable("11");
         ComboBoxFixture projectJdkComboBox = newProjectFirstPage.find(ComboBoxFixture.class, byXpath(XPathDefinitions.JDK_COMBOBOX), Duration.ofSeconds(10));
         String currentlySelectedProjectSdk = listOfRemoteTextToString(projectJdkComboBox.findAllText());
-        assertTrue(currentlySelectedProjectSdk.contains("8"), "Selected project SDK should be Java 8 but is '" + currentlySelectedProjectSdk + "'");
-        newProjectFirstPage.setProjectSdkIfAvailable("11");
+        assertTrue(currentlySelectedProjectSdk.contains("11"), "Selected project SDK should be Java 8 but is '" + currentlySelectedProjectSdk + "'");
```
Comment (olkornii) [inline src/test-project/src/test/java/com/redhat/devtools/intellij/commonuitest/fixtures/test/dialogs/project_manipulation/NewProjectDialogTest.java:339]: Rewrite comments "Selected Project blablalba 8 ..." to "11".

File: src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java [hunk 2]
```diff
@@ -241,7 +241,18 @@ private JButtonFixture welcomeFrameLink(String label) {
         if (UtilsKt.hasAnyComponent(this, byXpath(XPathDefinitions.RECENT_PROJECT_PANEL_NEW))) {
             return button(byXpath(XPathDefinitions.jBOptionButton(label)), Duration.ofSeconds(2));
         }
-        return button(byXpath(XPathDefinitions.nonOpaquePanel(label)), Duration.ofSeconds(2));
+        if (ideaVersion >= 20232) {
+            // code for IntelliJ Idea 2023.2+
+
+            try {
+                return button(byXpath("//div[@defaulticon='createNewProjectTab.svg']"), Duration.ofSeconds(2));
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java]: whole thing was redundant

File: src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java [hunk 3]
```diff
@@ -241,7 +241,18 @@ private JButtonFixture welcomeFrameLink(String label) {
...
+            } catch (WaitForConditionTimeoutException e) {
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java]: whole thing was redundant

File: src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java [hunk 4]
```diff
@@ -254,11 +256,20 @@ private void removeTopProjectFromRecentProjects() {
...
+        // Clicks on X on first recent project to remove it from the recent projects list (visible only when hovered over with cursor)
         recentProjects.runJs("const horizontal_offset = component.getWidth()-22;\n" +
                 "robot.click(component, new Point(horizontal_offset, 22), MouseButton.LEFT_BUTTON, 1);");
 
-        // Code for IntelliJ Idea 2020.3 or newer
-        if (ideaVersion >= 20203) {
+        if (ideaVersion >= 20232) {
+           ComponentFixture removeDialog = remoteRobot.find(ComponentFixture.class, byXpath(XPathDefinitions.MY_DIALOG), Duration.ofSeconds(2));
+            try {
+                removeDialog.findText("Remove From List")
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java]: fixed

File: src/test-project/src/test/java/com/redhat/devtools/intellij/commonuitest/fixtures/test/mainidewindow/toolwindowspane/ProjectExplorerTest.java [hunk 5]
```diff
@@ -118,8 +119,10 @@ public void openViewsPopupTest() {
     public void selectOpenedFileTest() {
         projectExplorer.openFile(PROJECT_NAME, PROJECT_NAME + ".iml");
         projectExplorer.projectViewTree().clickRow(0);
+        SharedSteps.waitForComponentByXpath(remoteRobot,3,1, byXpath(XPathDefinitions.MY_ICON_LOCATE_SVG));
```
Comment (olkornii) [inline src/test-project/src/test/java/com/redhat/devtools/intellij/commonuitest/fixtures/test/mainidewindow/toolwindowspane/ProjectExplorerTest.java:122]: Try to verify "...3,1..." pause works.

**Review by olkornii — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java [hunk 6]
```diff
@@ -156,24 +148,23 @@ public void openSettingsDialog() {
      * @return fixture for the 'Tip Of the Day' dialog
      */
     public TipDialog openTipDialog() {
-        if (ideaVersion <= 20202) {
+        if (ideaVersion >= 20211) {
+            FlatWelcomeFrame flatWelcomeFrame = remoteRobot.find(FlatWelcomeFrame.class, Duration.ofSeconds(2));
+            if (ideaVersion >= 20223) {         //  COMMUNITY_V_2022_3 and higher version have different labels for Learn button
```
[the remark this answers is not in the document]
Comment (martinszuc) [reply src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/FlatWelcomeFrame.java]: Fixed in open tip dialog commit. Also changed selectNewProjectTypeTest in same manner.

**Review by lgrossma — APPROVED** [review]

**Pull request merged**